docs/tutorials/dcdi/decorators/__init__.py

```diff
@@ -0,0 +1,99 @@
+"""
+
+A customer walks into a store. Do the steps to interact with them:
+
+- Get a correct greeter
+
+- Interact with them
+
+Simple wired application:
+
+- Settings that say what punctuation to use
+
+- Registry
+
+- A bundled Greeter and Customer
+
+- An add-on which defines a FrenchGreeter and FrenchCustomer
+
+- A Datastore which stores/retrieves instances of Customers
+
+- Request and View factories to assemble the full processing chain
+
+"""
+from typing import List
+
+import venusian
+
+from wired import ServiceRegistry, ServiceContainer
+from .models import (
+    Customer, Datastore, Greeter, Resource, Request, Settings, Url, View
+)
+from .utils import (
+    process_request,
+    register_dataclass)
+
+
+def app_bootstrap(settings: Settings) -> ServiceRegistry:
+    # Make the registry
+    registry = ServiceRegistry()
+
+    # Make a container to use during the initialization phase
+    container: ServiceContainer = registry.create_container()
```

Comment:
can this similarly be removed? I don't see many places where the venusian callback is using `scanner.container`.

Reply:
In my wired.dataclasses branch right now I'm actually documenting the use of container as an injectable.

Reply:
that's not a config-time container though right? That's the container invoked via a factory?

Reply:
Can you send me a link to it?

Reply:
Here is [the branch](https://github.com/pauleveritt/wired/tree/dataclasses)

Here is [a test for injecting the container](https://github.com/pauleveritt/wired/blob/dataclasses/tests/dataclasses/integration/test_containers.py)

The idea is that `wired` might have some operation that `wired.dataclasses` doesn't have the declarative DI setup for. E.g. you just added `container.register_singleton`. Allowing `wired.dataclasses` to get the `container` as an injectable then lets me use `__post_init__` to go grab something in the container.

Reply:
Ok I see what you're saying and support that use case of injecting the container. I think that's different than the config-time container that I'm talking about here in `app_bootstrap`. The container here is the same one we were talking about in `setup(registry, datastore)`.

Reply:
Actually you're right. That container is provided in the Scanner but isn't used. I'll remove it. the container I'm referring to -- the one that can be injected -- needs to be a per-request container.

Reply:
I wasn't clear at the start but I'm proposing to get rid of the container entirely in this function, not just in the callback.

Side note, I don't see anything else in this PR so if that gets fixed are you ready for me to merge it?

Reply:
It's used several lines down to get the datastore.

Reply:
Ok I see what you're saying. So the datastore is registered via the scan, then you create a container here to grab it and pass it around to finish the config. That's a bit roundabout to me but I can live with it. I'm happy to merge this if you're ready?

Reply:
Yep, let's merge. Thanks!